### DISCUSSION TITLE
Heads up: require('chromedriver/install') behavior changed in 122

### DISCUSSION THREAD

**Discussion by joshden** [giggio/node-chromedriver#498] [General]
We've been using in some of our projects a update-chromedriver.js file with these contents:

    process.env.DETECT_CHROMEDRIVER_VERSION = "true"
    require("chromedriver/install")

We did like this so we could explicitly set the DETECT_CHROMEDRIVER_VERSION environment variable and call this script from package.json, which doesn't have a native cross-platform way to set environment variables. 

We were using version 121 of this package. It correctly detects our Chrome version and downloads the corresponding chromedriver package. However, when trying to run this same thing with the latest version, it is doing nothing. I tracked it down to this commit, starting in version 122 of this package: https://github.com/giggio/node-chromedriver/commit/c91fa59595ca031d517e5ae41fbad2b002196e89.

I realize we were taking advantage of this undocumented behavior, and it was changed to be able to unit test this file. Here are a few alternatives to work with this change:

1. Run chromedriver/install with fork

        const { fork } = require('child_process')
        process.env.DETECT_CHROMEDRIVER_VERSION = "true"
        fork('node_modules/chromedriver/install')

2. require chromedriver/install and call .install()

        process.env.DETECT_CHROMEDRIVER_VERSION = "true"
        process.env.NODE_ENV = "test"
        const Installer = require('chromedriver/install')
        new Installer().install()

3. Resort to using cross-env in package.json to set environment variable in a script:

        "update-chromedriver": "cross-env DETECT_CHROMEDRIVER_VERSION=true node node_modules/chromedriver/install.js"

Would be interested if anyone has a more elegant solution to this new behavior, and I hope this information could be helpful to any others who were relying on this previous behavior. Thanks.

**Comment by giggio** [comment]
If the intent is to run an npm action I'd use option 3, as I've done many times in the past.
